clippy_lints/src/utils/mod.rs

```diff
@@ -866,6 +867,14 @@ pub fn return_ty<'tcx>(cx: &LateContext<'tcx>, fn_item: hir::HirId) -> Ty<'tcx>
     cx.tcx.erase_late_bound_regions(&ret_ty)
 }
 
+/// Walk into `ty` and returns `true` if any inner type is the same as `other_ty`
+pub fn contains_ty<'tcx>(ty: Ty<'tcx>, other_ty: Ty<'tcx>) -> bool {
```

Comment:
```suggestion
pub fn contains_ty(ty: Ty<'_>, other_ty: Ty<'_>) -> bool {
```
We can elide the lifetimes here for simplicity